File: new-exchange/app/infrastructure/db/database.config.go
package db

import (
	"fmt"
	"gorm.io/driver/postgres"
	"gorm.io/gorm"
)

type DbConfig struct {
	Host     string
	Port     int
	User     string
	Password string
	DbName   string
}

func ConnectToDb(config DbConfig) *gorm.DB {
	dsn := fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%d",
		config.Host, config.User, config.Password, config.DbName, config.Port)
	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
	if err != nil {
		panic("failed to connect database")
	}
	return db
}

File: new-fido/src/services/database/database.config.ts
import { registerAs } from "@nestjs/config";
import { ParcelDeliveryEntity } from "../../parcel-delivery/infrastructure/entity/parcel-delivery";

export const DbConfig = registerAs("database", () => ({
  type: "postgres",
  host: process.env.DB_HOST || "localhost",
  port: parseInt(process.env.DB_PORT, 10) || 5433,
  username: process.env.DB_USER || "stroka01",
  password: process.env.DB_PASSWORD || "user",
  database: process.env.DB_NAME || "dev_db",
  entities: [ParcelDeliveryEntity],
  synchronize: process.env.NODE_ENV === "development",
  logging: process.env.NODE_ENV === "development",
  migrations: [`${__dirname}/../../db/migrations/*{.ts,.js}`],
  migrationsTableName: "migrations",
}));
